universe/Universe.cpp

```diff
@@ -1660,6 +1660,21 @@ void Universe::ForgetKnownObject(int empire_id, int object_id) {
     objects.Remove(object_id);
 }
 
+void Universe::CheckShipDesignVisibility(int empire_id, int object_id, Visibility vis) {
```

Comment:
could use a better name, to clarify that it's setting, not just "checking" the visibility...